modules/mbedtls/configs/config-tls-generic.h

```diff
@@ -482,12 +482,13 @@
 #define MBEDTLS_PSA_P256M_DRIVER_ENABLED
 #endif
 
-#if defined(CONFIG_ARCH_POSIX) && !defined(CONFIG_PICOLIBC)
+#if defined(CONFIG_ARCH_POSIX) && !defined(CONFIG_PICOLIBC) && !defined(CONFIG_SECURE_STORAGE)
 #define MBEDTLS_PSA_ITS_FILE_C
+#define MBEDTLS_PSA_CRYPTO_STORAGE_C
```

Comment:
As a temporary solution, this is Ok. However, `MBEDTLS_PSA_CRYPTO_STORAGE_C` shouldn't be blindly enabled without SETTINGS presence consideration. It creates misconfiguration in subsystems.

Reply:
This behavior has been present for a while, and indeed should be changed (though not based solely on `SETTINGS` as this is just one possibility for persistent storage).